### PR TITLE
[one-cmds] Introduce workflow

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit introduces workflow to one-cmds. You can see the usages in the test of below draft.

Related: #9192 
Draft: #9354 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Review by jyoungyun — APPROVED** [review]
LGTM

File: compiler/one-cmds/onelib/TopologicalSortHelper.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def sort_util(self, v, visited, stack):
+        visited[v] = True
+
+        for i in self.graph[v]:
+            if visited[i] == False:
+                self.sort_util(i, visited, stack)
+
+        stack.insert(0, v)
+
+    def sort(self):
```
Comment (jinevening) [inline compiler/one-cmds/onelib/TopologicalSortHelper.py:37]: Better to handle cycle dependency, e.g., throw exception.

File: compiler/one-cmds/onelib/OptionBuilder.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            if k in optional and v == "True":
+                options.extend(['--' + k])
+                continue
+            options.extend(['--' + k, v])
+        return options
+
+    def _build_optimize(self, commands):
+        options = []
+        required = ['input_path', 'output_path']
+        optional = ['generate_profile_data', 'change_outputs']
```
Comment (jinevening) [inline compiler/one-cmds/onelib/OptionBuilder.py:54]: It seems that `change_outputs` should be called with value, e.g., `--change_outputs XXX`.

The variable names `required` and `optional` are a bit confusing to me. I think the below naming is what you intended. Please correct me if I'm wrong.

`required` -> `opt_with_val`
`optional` -> `opt_without_val`

File: compiler/one-cmds/onelib/WorkflowRunner.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            if not self.WORKFLOW_STEPS_K in json_contents[workflow_k] and not self.CFG_REFERENCE_K in json_contents[workflow_k]:
+                raise ValueError("Each workflow should have either \"" +
+                                 self.WORKFLOW_STEPS_K + "\" or \"" +
+                                 self.CFG_REFERENCE_K + "\"")
+        for workflow_k in workflows:
+            if self.WORKFLOW_STEPS_K in json_contents[workflow_k] and self.CFG_REFERENCE_K in json_contents[workflow_k]:
+                raise ValueError("\"" + self.WORKFLOW_STEPS_K + "\" and \"" +
+                                 self.CFG_REFERENCE_K + "\" are exclusive key")
+
+        # each step should have ONE_CMD_TOOK_K and COMMANDS_K
```
Comment (jinevening) [inline compiler/one-cmds/onelib/WorkflowRunner.py:69]:
```suggestion
        # each step should have ONE_CMD_TOOL_K and COMMANDS_K
```

File: infra/debian/compiler/one-compiler.install [hunk 4]
```diff
@@ -28,6 +28,10 @@ usr/bin/one-quantize usr/share/one/bin/
 usr/bin/one-version usr/share/one/bin/
 usr/bin/onelib/constant.py usr/share/one/bin/onelib/
 usr/bin/onelib/make_cmd.py usr/share/one/bin/onelib/
+usr/bin/onelib/CfgRunner.py usr/share/one/bin/onelib/
```
Comment (seanshpark) [inline infra/debian/compiler/one-compiler.install:31]: please split this from `compiler/one-cmds` files.

File: infra/debian/compiler/one-compiler.install [hunk 5]
```diff
@@ -28,6 +28,10 @@ usr/bin/one-quantize usr/share/one/bin/
...
+usr/bin/onelib/OptionBuilder.py usr/share/one/bin/onelib/
+usr/bin/onelib/TopologicalSortHelper.py usr/share/one/bin/onelib/
+usr/bin/onelib/WorkflowRunner.py usr/share/one/bin/onelib/
```
[the remark this answers is not in the document]
Comment (mhs4670go) [reply infra/debian/compiler/one-compiler.install:34]: You're right. I just followed the name of file contents. Class starts from capital letter and function doesn't. As you said, we need a discussion for this.

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

I think you should update the debian related file to include newly introduced files.

**Comment by seanshpark** [comment]
I'll land this when debian packaged related PR is ready (not to break debian package build)

**Pull request merged**